internal/services/mssql/mssql_database_resource_test.go

```diff
@@ -1020,6 +1037,22 @@ resource "azurerm_mssql_database" "test" {
 `, r.template(data), data.RandomInteger)
 }
 
+func (r MsSqlDatabaseResource) basicWithSku(t *testing.T, data acceptance.TestData, sku string) string {
+	if _, err := mssqlValidate.DatabaseSkuName()(sku, ""); err != nil {
+		t.Errorf("%s", errors.Join(err...).Error())
+	}
```

Comment:
Made this `basicWithSku` method accept a `sku` string, to allow testing basic, default configurations with different skus easy in the future. This does however require passing a `*testing.T` through to the method. Happy to change if you'd rather see another approach used.

Reply:
Appreciate the thought to make it easier and more robust to test the basic config for more skus!

My thoughts on this are that we don't typically test more than a few skus to ensure that they function so I don't see a _huge_ need for this. We do prefer using explicit configs where possible rather than parameterised and it is non-standard to validate testing configs in the provider. All up this would be introducing a new "inconsistency" into the provider, which given the size of the code base and the many contributors we have, we try not to do since it makes keeping standard practices and reviews more difficult.

Given the reasons above I would prefer to see this validation in the config removed, the test case is perfectly valid though.

Reply:
Thank you for the feedback and makes sense @stephybun. I originally had the test configuration explicit. Had a gut feeling that would be preferred; a large part of the reason I called this bit out via comment. I should have listed to my instinct, keep it simple 🙃

An easy change, and I will fixup as you've instructed. Cheers.